File: ProjetSopraFinal/src/main/java/com/example/demo/model/Salle.java
package com.example.demo.model;

import java.util.Set;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("S")
public class Salle extends Materiel {

	private Integer nbDePersonnes;
	private Set<Module> modules;

	public Salle() {
	}

	public Integer getNbDePersonnes() {
		return nbDePersonnes;
	}

	public void setNbDePersonnes(Integer nbDePersonnes) {
		this.nbDePersonnes = nbDePersonnes;
	}

	@OneToMany (mappedBy = "salle")
	public Set<Module> getModules() {
		return modules;
	}

	public void setModules(Set<Module> modules) {
		this.modules = modules;
	}
	
	
	
}

File: AngularSopraFinal/src/app/model/salle.ts
import {Module} from './module';

export class Salle {

  constructor(private _nbDePersonne?, private _modules?) {
  }

  public get nbDePersonne(): number {
    return this._nbDePersonne
  }
  public set nbDePersonne(_nbDePersonne: number) {
    this._nbDePersonne = _nbDePersonne;
  }

  public get modules(): Array<Module> {
    return this._modules;
  }
  public set modules(_modules: Array<Module>) {
    this._modules = _modules;
  }
}